### PR TITLE
EN-7793-view-func-bls-keys-status

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Added a view function for the auctions system sc that returns (blsKey - status) pairs.

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: vm/systemSmartContracts/auction.go [hunk 1]
```diff
@@ -1386,6 +1388,50 @@ func (s *stakingAuctionSC) EpochConfirmed(epoch uint32) {
...
+
+	for _, blsKey := range registrationData.BlsPubKeys {
+		vmOutput, err := s.executeOnStakingSC([]byte("getBLSKeyStatus@" + hex.EncodeToString(blsKey)))
+		if err != nil {
+			s.eei.AddReturnMessage("cannot get bls key status: bls key - " + hex.EncodeToString(blsKey) + " error - " + err.Error())
+			continue
+		}
+
+		if vmOutput.ReturnCode != vmcommon.Ok {
+			continue
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/auction.go:1420]: Do we not need also a s.eei.AddReturnMessage here?

Comment (sasurobert) [reply vm/systemSmartContracts/auction.go:1420]: should not happen.

File: vm/systemSmartContracts/auction.go [hunk 2]
```diff
@@ -1386,6 +1388,50 @@ func (s *stakingAuctionSC) EpochConfirmed(epoch uint32) {
...
+		}
+
+		if len(vmOutput.ReturnData) != 1 {
+			s.eei.AddReturnMessage("could not get bls key status for key " + hex.EncodeToString(blsKey))
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply vm/systemSmartContracts/auction.go]: Done.

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
👍

**Pull request merged**